### ISSUE TITLE
Cannot mount second hard drive after ChimeraOS 40 update

### ISSUE THREAD

**Issue by photoncody** [ChimeraOS/chimeraos#552]
After the update, I noticed that my second hard drive would not mount.

I tried a format in the Chimera web interface, and just got this error:

![image](https://user-images.githubusercontent.com/65925838/228988532-518f6c30-6799-488e-94d1-c6f45677c100.png)

I tried multiple reboots and even drilled down using the commands it gives me and it looks like the drive is failing to mount with this error code:

```
ConsolePC mount-media.sh[12025]: Error mounting /dev/sdb1: wrong fstype:  - {
ConsolePC mount-media.sh[12025]:       "path": "/dev/sdb1",
ConsolePC mount-media.sh[12025]:       "label": null,
ConsolePC mount-media.sh[12025]:       "fstype": null
ConsolePC mount-media.sh[12025]:  }
ConsolePC systemd[1]: media-mount@sdb1.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
...
ConsolePC mount-media.sh[12047]: Error mounting /dev/sdb1 (status = 1)
ConsolePC systemd[1]: media-mount@sdb1.service: Main process exited, code=exited, status=1/FAILURE
```

I used sudo to try and run the command that the service was trying to run:
`sudo /usr/lib/media-support/mount-media.sh add sdb1`

and got the error code:
```
Call failed: Not authorized to perform operation
Error mounting /dev/sdb1 (status = 1)
```

I ran the above command as sudo so I'm not sure why it would be unauthorized.

Let me know if you need me to do anything.

**Comment by pastaq** [comment]
This is a known issue and we're working on a hotfix for it. A new version will be released shortly. Thank you for reporting it,

**Comment by photoncody** [comment]
Looks like the issue is still happening.

![image](https://user-images.githubusercontent.com/65925838/229017606-6771844b-ae2c-4085-b633-2933abefc802.png)

It gets past the issue on this screen, but half of the time it still has the same error from the previous post.

I fixed it by going into disks, manually mounting the drive as /run/media/sdb, and then taking ownership of the disk with gamer user.

**Comment by pastaq** [comment]
It seems like there's a separate issue with the web ui for this. It would be better to use the CLI as that will make changes that persist after reboot. I reproduced your issue but had no problems when I did it manually:
`/usr/bin/shadowblip/format-media --full --device /dev/sda`

**Comment by Rahzadan** [comment]
I'm having this same issue since updating past ChimeraOS 40. I lost my mounted internal HDD's after this update.
@pastaq , trying your workaround from the CLI did seem to format the drives, but after rebooting, they still don't show up in Gamepad UI anymore, and I can't install games to them. Trying to format them via WebUI also results in the format log the same as above users.

GamePad UI can only see my Internal drive that ChimeraOS is installed on.

**Comment by pastaq** [comment]
This should fix it. Please give it a try.
```
sudo frzr unlock
git clone https://github.com/ShadowBlip/steam-removable-media.git && cd steam-removable-media
sudo ./install.sh
reboot
```

**Comment by Rahzadan** [comment]
![image](https://user-images.githubusercontent.com/28596721/230206384-967da751-65b4-465d-9171-9f01f2150ffd.png)

**Comment by pastaq** [comment]
> frzr command not found.
> 
> ![image](https://user-images.githubusercontent.com/28596721/230206384-967da751-65b4-465d-9171-9f01f2150ffd.png)

There was a typo. Try again with the updated instructions.

**Comment by Rahzadan** [comment]
@pastaq, Still not working unfortunately. I was able to successfully run the corrected commands you mentioned above, but after rebooting, my drives were still not present.

I tried re-formatting again from the WebUI and got the following log:

> 1024+0 records in
1024+0 records out
524288 bytes (524 kB, 512 KiB) copied, 0.00904756 s, 57.9 MB/s
mke2fs 1.47.0 (5-Feb-2023)
Job for media-mount@sda1.service failed because the control process exited with error code.
See "systemctl status media-mount@sda1.service" and "journalctl -xeu media-mount@sda1.service" for details.

The 'Storage' page on my WebUI looks like this. It just says the drives are mounted on "quote":
![image](https://user-images.githubusercontent.com/28596721/230385347-aae50fdf-29de-41c5-94a1-9354fa256aea.png)

**Comment by scamfield** [comment]
Hi @photoncody @Rahzadan, I encountered the same problem with my second drive not mounting after upgrading from version 39 to 41. I'm not certain if this solution will work for you, but it did for me. Upon examining the logs, I noticed the following error during boot after the upgrade:

```
Apr 07 13:06:18 SteamConsole mount-media.sh[709]: Call failed: Not authorized to perform operation
Apr 07 13:06:18 SteamConsole systemd-journald[452]: /var/log/journal/ccf1a74e9e8d4f8abc4d5ca5610d43f2/user-1000.journal: Monotonic clock jumped backwards relat>
Apr 07 13:06:18 SteamConsole systemd[1]: run-u2.service: Main process exited, code=exited, status=1/FAILURE
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░
░░ An ExecStart= process belonging to unit run-u2.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
Apr 07 13:06:18 SteamConsole systemd[1]: run-u2.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░
░░ The unit run-u2.service has entered the 'failed' state with result 'exit-code'.
Apr 07 13:06:18 SteamConsole mount-media.sh[633]: Error mounting /dev/sdb1 (status = 1)
Apr 07 13:06:18 SteamConsole systemd[1]: media-mount@sdb1.service: Main process exited, code=exited, status=1/FAILURE
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
░░
░░ An ExecStart= process belonging to unit media-mount@sdb1.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
Apr 07 13:06:18 SteamConsole systemd[1]: media-mount@sdb1.service: Failed with result 'exit-code'.
```

When I checked the service, it showed that my sdb1 had failed:

```
[gamer@SteamConsole ~]$ sudo service media-mount@sdb1.service status
sudo: service: command not found
[gamer@SteamConsole ~]$ systemctl status media-mount@sdb1.service
× media-mount@sdb1.service - Mount removable media on sdb1
     Loaded: loaded (/usr/lib/systemd/system/media-mount@.service; static)
     Active: failed (Result: exit-code) since Fri 2023-04-07 13:26:36 IST; 2h 10min ago
    Process: 626 ExecStart=/usr/lib/media-support/mount-media.sh add sdb1 (code=exited, status=1/FAILURE)
   Main PID: 626 (code=exited, status=1/FAILURE)
        CPU: 70ms

Apr 07 13:26:36 SteamConsole systemd[1]: Starting Mount removable media on sdb1...
Apr 07 13:26:36 SteamConsole mount-media.sh[690]: Running as unit: run-u3.service
Apr 07 13:26:36 SteamConsole mount-media.sh[626]: Error mounting /dev/sdb1 (status = 1)
Apr 07 13:26:36 SteamConsole systemd[1]: media-mount@sdb1.service: Main process exited, code=exited, status=1/FAILURE
Apr 07 13:26:36 SteamConsole systemd[1]: media-mount@sdb1.service: Failed with result 'exit-code'.
Apr 07 13:26:36 SteamConsole systemd[1]: Failed to start Mount removable media on sdb1.
```

After running the script manually and investigating the issue, I thought it might be related to the user not having the proper permissions to mount:

`sudo usermod -a -G storage gamer`

However, the error persisted. I then considered the possibility of missing polkit policy:

I created the following file and its contents:

/etc/polkit-1/rules.d/50-mount.rules:

```
polkit.addRule(function(action, subject) {
    if (action.id.match("org.freedesktop.udisks2.") &&
        subject.user == "gamer") {
        return polkit.Result.YES;
    }
});
```

I restarted the polkit service and verified the logs:

```
sudo systemctl restart polkit

journalctl -b -u polkit
```

After running the mount command again, I encountered a different error. Upon inspecting the symbolic links, it seemed like the issue was related to the mount previously using the old device name rather than the UUID. I removed the old link and ran the command once more, which resolved the issue:


```
[gamer@SteamConsole ~]$ sudo /usr/lib/media-support/mount-media.sh add sdb1
Running as unit: run-u57.service
**** Mounted /dev/sdb1 at /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0 ****
ln: failed to create symbolic link '/run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0/SteamLibrary': File exists

[gamer@SteamConsole ~]$ ls -lah /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0/
total 12K
drwxr-xr-x  3 gamer gamer 4.0K Apr  7 15:58 .
drwxr-x---+ 3 root  root    60 Apr  7 16:00 ..
-rwxr-xr-x  1 gamer gamer   68 Apr  3 11:16 libraryfolder.vdf
drwxr-xr-x  5 gamer gamer 4.0K Apr  7 08:35 steamapps
lrwxrwxrwx  1 root  root    15 Apr  3 11:16 SteamLibrary -> /run/media/sdb1

[gamer@SteamConsole ~]$ sudo rm  /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0/SteamLibrary

[gamer@SteamConsole ~]$ sudo /usr/lib/media-support/mount-media.sh add sdb1
Running as unit: run-u65.service
**** Mounted /dev/sdb1 at /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0 ****
Running as unit: run-u36.service
Finished with result: success
Main processes terminated with: code=exited/status=0
Service runtime: 4ms
CPU time consumed: 4ms
Sent URL to steam: steam://addlibraryfolder/%2f%72%75%6e%2f%6d%65%64%69%61%2f%67%61%6d%65%72%2f%33%32%33%66%62%33%65%31%2d%64%64%39%63%2d%34%65%64%64%2d%38%37%30%30%2d%35%36%36%31%39%62%61%32%64%65%65%30
/dev/sdb1 added as a steam library at /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0
```

After several reboots, everything is working for me and mounting as expected during boot. This may or may not be the solution for you, but it's how I fixed my mounting issue.

**Comment by pastaq** [comment]
> 
Most of these issues are resolved in the latest version from my comment above so it won't be necessary to do most of it. The polkit was fixed and we no longer create that link as it was causing issues and not actually solving the problem it was intended to fix. Also, the mount lock was fixed that was causing the mount to start too early during formatting. I think the only thing that isn't covered is if the link already exists it is causing issues. I'll look into this to see if there is a way to clean up those links. Thanks.

**Comment by pastaq** [comment]
@scamfield I updated the above instructions based on your findings, thanks!

**Comment by Rahzadan** [comment]
I've spent a stupid amount of time the last couple of days trying to get my drives mounted again. Absolutely nothing in this thread has worked. I even tried deleting all partitions on all drives and re-installing ChimeraOS.

First off, the **2023-03-28 (d13bf66)** installer iso is broken. It gives an "bootstrap failed" error when trying to install. I had to use **2023-02-15 (432f20f)** - which did download the latest ChimeraOS 41 image during install.

- WIth a fresh ChimeraOS 41 installed, I tried to use the WebUI to format, which, again, fails with the output from OP.
- Tried **/usr/bin/shadowblip/format-media --full --device /dev/sda** again, which resulted in format failure.
- Tried `sudo frzr-unlock
git clone https://github.com/ShadowBlip/steam-removable-media.git && cd steam-removable-media
sudo ./install.sh
reboot` which did some stuff and had no errors, but it didn't do anything to make my drives show, even after a reboot.

- Tried `sudo mkdir /run/media/sda1
sudo mount /dev/sda /run/media/sda1
sudo rm /run/media/sda1/SteamLibrary
sudo umount /run/media/sda1
sudo /usr/lib/media-support/mount-media.sh add sdb1` however on the 2nd step to mount /dev/sda it enountered an error (same as OP) and would not mount.

I'm now at a complete loss, and it seems like something between v39 and v41 has completely broken secondary drives.

**Comment by scamfield** [comment]
Hi @Rahzadan,  I'm not sure if you've already tried this in your update, but have you tested the polkit policy I mentioned earlier [here](https://github.com/ChimeraOS/chimeraos/issues/552#issuecomment-1500437316) , as well as the steps I took to remove the old SteamLibrary reference? It seems that there may be an issue with the polkit policy on some systems, including mine, and following those steps resolved the problem for me. I don't think these solutions are included in the provided fixes.

**Comment by pastaq** [comment]
Tried sudo mkdir /run/media/sda1 sudo mount /dev/sda /run/media/sda1 sudo rm /run/media/sda1/SteamLibrary sudo umount /run/media/sda1 sudo /usr/lib/media-support/mount-media.sh add sdb1 however on the 2nd step to mount /dev/sda it enountered an error (same as OP) and would not mount.

Shoot, I had a typo in those commands. The step that failed should have been  `sudo mount /dev/sda1 /run/media/sda1` it was missing the 1 after the first sda. I have updated the instructions above.

**Comment by pastaq** [comment]
Okay, looks like we found the missing rule. Internal drives need the filesystem-mount-system permission as well, which is why it was working for SD cards and USB disks, but not SATA and NVME. I verified this on my own machine after installing an additional internal drive.
This commit https://github.com/ShadowBlip/steam-removable-media/commit/86b1b69f927a440650b79be5c7d5f662e08bc2fa should solve this issue once and for all. Please test by installing the latest version per the instructions above.

**Comment by scamfield** [comment]
Hello @pastaq  - I can verify that the previously missing policy is now functioning! After executing the install.sh script and restarting, the second drive was correctly mounted. However, I had to unmount it and employ the add script to re-add the device, enabling it to show up within Steam. This is a minor concern that could potentially be resolved by updating the instructions. Here is the output of the commands I executed.

```
[gamer@SteamConsole ~]$ sudo frzr-unlock

We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

For security reasons, the password you type will not be visible.

[sudo] password for gamer:
:: Synchronizing package databases...
 core                                                                153.9 KiB   290 KiB/s 00:01 [########################################################] 100% extra                                                              1748.8 KiB  2.80 MiB/s 00:01 [########################################################] 100% community                                                             7.3 MiB  6.98 MiB/s 00:01 [########################################################] 100% multilib                                                            161.9 KiB   274 KiB/s 00:01 [########################################################] 100%frzr deployment chimeraos-41_453df5f unlocked
[gamer@SteamConsole ~]$ git clone https://github.com/ShadowBlip/steam-removable-media.git && cd steam-removable-media
Cloning into 'steam-removable-media'...
remote: Enumerating objects: 264, done.
remote: Counting objects: 100% (63/63), done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 264 (delta 40), reused 52 (delta 34), pack-reused 201
Receiving objects: 100% (264/264), 52.25 KiB | 1.54 MiB/s, done.
Resolving deltas: 100% (121/121), done.
[gamer@SteamConsole steam-removable-media]$ sudo ./install.sh
Installing Media Support Package
'usr/bin/shadowblip/format-media' -> '/usr/bin/shadowblip/format-media'
'usr/bin/steamos-polkit-helpers/steamos-format-sdcard' -> '/usr/bin/steamos-polkit-helpers/steamos-format-sdcard'
'usr/bin/steamos-polkit-helpers/steamos-retrigger-automounts' -> '/usr/bin/steamos-polkit-helpers/steamos-retrigger-automounts'
'usr/bin/steamos-polkit-helpers/steamos-trim-devices' -> '/usr/bin/steamos-polkit-helpers/steamos-trim-devices'
'usr/lib/hwsupport/format-sdcard.sh' -> '/usr/lib/hwsupport/format-sdcard.sh'
'usr/lib/hwsupport/trim-devices.sh' -> '/usr/lib/hwsupport/trim-devices.sh'
'usr/lib/media-support/format-media.sh' -> '/usr/lib/media-support/format-media.sh'
'usr/lib/media-support/mount-media.sh' -> '/usr/lib/media-support/mount-media.sh'
'usr/lib/systemd/system/media-mount@.service' -> '/usr/lib/systemd/system/media-mount@.service'
'usr/lib/udev/rules.d/99-media-mount.rules' -> '/usr/lib/udev/rules.d/99-media-mount.rules'
'usr/share/polkit-1/actions/org.shadowblip.media-support.policy' -> '/usr/share/polkit-1/actions/org.shadowblip.media-support.policy'
'usr/share/polkit-1/rules.d/org.shadowblip.media-support.rules' -> '/usr/share/polkit-1/rules.d/org.shadowblip.media-support.rules'
[gamer@SteamConsole steam-removable-media]$ reboot
```

After the reboot:

```
[gamer@SteamConsole ~]$ df -h
Filesystem      Size  Used Avail Use% Mounted on
dev             7.8G     0  7.8G   0% /dev
run             7.8G  1.1M  7.8G   1% /run
/dev/sda2       112G  7.0G  103G   7% /
/dev/sda2       112G  7.0G  103G   7% /frzr_root
overlay         112G  7.0G  103G   7% /etc
tmpfs           7.8G  120M  7.7G   2% /dev/shm
tmpfs           7.8G   20K  7.8G   1% /tmp
/dev/sda2       112G  7.0G  103G   7% /var
/dev/sda2       112G  7.0G  103G   7% /home
/dev/sda1       511M  104M  408M  21% /boot
/dev/sdb1       458G  323G  136G  71% /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0
tmpfs           1.6G   28K  1.6G   1% /run/user/1000
[gamer@SteamConsole ~]$ sudo rm /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0/SteamLibrary
[sudo] password for gamer:
rm: cannot remove '/run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0/SteamLibrary': Is a directory
[gamer@SteamConsole ~]$ sudo rm -Rf /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0/SteamLibrary
[gamer@SteamConsole ~]$ sudo /usr/lib/media-support/mount-media.sh add sdb1
Running as unit: run-u48.service
Call failed: Device /dev/sdb1 is already mounted at `/run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0'.
Error mounting /dev/sdb1 -- (status = 1)
--- 0 ---
[gamer@SteamConsole ~]$
```

Unmounted the volume and re-added to get it to appear within steam:

```
[gamer@SteamConsole ~]$ sudo umount /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0
[gamer@SteamConsole ~]$ sudo /usr/lib/media-support/mount-media.sh add sdb1
Running as unit: run-u55.service
Mounted /dev/sdb1 at /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0
Desktop Libray not found. Creating...
Adding symlink to steamapps dir
Adding symlink to libraryfolder.vdf
Setting user permissions for /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0...
Sending URL to steam: steam://addlibraryfolder/%2f%72%75%6e%2f%6d%65%64%69%61%2f%67%61%6d%65%72%2f%33%32%33%66%62%33%65%31%2d%64%64%39%63%2d%34%65%64%64%2d%38%37%30%30%2d%35%36%36%31%39%62%61%32%64%65%65%30
Running as unit: run-u36.service
Finished with result: success
Main processes terminated with: code=exited/status=0
Service runtime: 3ms
CPU time consumed: 3ms
/dev/sdb1 added as a steam library at /run/media/gamer/323fb3e1-dd9c-4edd-8700-56619ba2dee0
[gamer@SteamConsole ~]$
```

**Comment by Rahzadan** [comment]
@pastaq @scamfield 

I was able to get my external drives working again. Since the commit to steam-removable-media mentioned above, I was able to format using the WebUI, but only AFTER installing the updated package using the instructions above. Seems it was indeed a problem with the removable media script.

The only (minor) issue I'm having now is the drives have their ugly UUID names within GamePad UI:

![image](https://user-images.githubusercontent.com/28596721/231243080-aa0d6f2d-cdbc-4fb4-9dd3-031eb6ade812.png)

Does anyone know how to change them to something more friendly like "Internal 2" "Internal 3" etc.

**Comment by pastaq** [comment]
> @pastaq @scamfield
> 
> I was able to get my internal SATA drives working again. Since the commit to steam-removable-media mentioned above, I was able to format using the WebUI, but only AFTER installing the updated package using the instructions above. Seems it was indeed a problem with the removable media script. The drives also remain mounted & present after a reboot.
> 
> The only (minor) issue I'm having now is the drives have their ugly UUID names within GamePad UI:
> 
> ![image](https://user-images.githubusercontent.com/28596721/231243080-aa0d6f2d-cdbc-4fb4-9dd3-031eb6ade812.png)
> 
> Does anyone know how to change them to something more friendly like "Internal 2" "Internal 3" etc.

As far as I'm aware we're stuck with that. Steam just pulls the mount path and udisks2 mounts by UUID for consistency.

**Comment by Rahzadan** [comment]
With ChimeraOS v39 and before, I remember they had much shorter names and weren't using the UUID. What was the exact reason for the change to UUIDs? I'm just curious.. Does it have something to do with the newer bootloader in v41+?

**Comment by pastaq** [comment]
> With ChimeraOS v39 and before, I remember they had much shorter names and weren't using the UUID. What was the exact reason for the change to UUIDs? I'm just curious.. Does it have something to do with the newer bootloader in v41+?

There were two main reasons for the changes.

1.) Users we're using drives in ways that weren't initially anticipated. Originally this was intended as a way to add install locations for steam games and no other use case was expected. Things like EmuDeck didn't exist yet or hadn't been used in ChimeraOS and we expected folks to use the Chimera app for ROM installs. TV/Movie support is limited as well so that wasn't really a consideration.

2.) In v39 and previous we were using root to mount and then chown the mounted directory's contents to the user. Since we were doing it manually it would sometimes conflict with udisks2, and wouldn't show up in the desktop easily really ever. Additionally, in v39 specifically one of the gnome packages dropped gnome-disks as an explicit dependency and we didn't notice until it was too late. We had been instructing users to use gnome-disks to access the drives in desktop mode if needed for a couple of versions. This led to an issue where some users we're able to manage their ROM's in the desktop since they couldn't manually add the drives and it didn't auto-populate. The steam library would still work, as well as any existing ROM's, but no new ROM's could be added unless the CLI was used, and it was complex.


This new method guarantees that users will always have full access to any drives they insert in the desktop as they are now explicitly mounted by the user account using udisks2. We do a couple extra things like add the steam apps folder(steam won't make this folder automatically if the root of the drive has other folders) as well as adding the SteamLibrary folder and symlink (steam desktop uses SteamLibrary as the top level dir with steam apps inside it, that would mean you'd need to install a game twice on the same drive if you wanted to use it in desktop mode and in "deck" mode)

Now that it's working it should (hopefully) alleviate any more of these edge cases where people want to use drives for EmuDeck or movies or file transfers, whatever, and we don't require users to manually re-mount in the desktop anymore to make these changes.

**Comment by ruineka** [comment]
> With ChimeraOS v39 and before, I remember they had much shorter names and weren't using the UUID. What was the exact reason for the change to UUIDs? I'm just curious.. Does it have something to do with the newer bootloader in v41+?

It's because we started using udisksctl to make the mounted drives match what Gnome expects to eliminate the many potential issues we were seeing users have.

**Comment by daniellasalle** [comment]
This problem is the same on a Atari VCS with the native eMMC drive and an additionnal NVMe drive.

If I install on the native eMMC drive and check under "STORAGE" the 2nd larger drive is not shown.

If I install on the bigger 2nd larger drive and then check under "STORAGE" the 1st smaller eMMC drive is not shown.

**Comment by pastaq** [comment]
> This problem is the same on a Atari VCS with the native eMMC drive and an additionnal NVMe drive installing the latest chimeraos-2023.07.06-x86_64.iso
> 
> If I install on the native eMMC drive and check under "STORAGE" the 2nd larger drive is not shown.
> 
> If I install on the bigger 2nd larger drive and then check under "STORAGE" the 1st smaller eMMC drive is not shown.

What is the output of `lsblk` and `df -h` for your system with the second drive installed?

**Comment by daniellasalle** [comment]
`[gamer@chimeraos 6cc09f63-45e7-4028-9243-3b64b8028dfa]$ lsblk

NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda            8:0    0   1.8T  0 disk 
├─sda1         8:1    0   511M  0 part /frzr_root/boot
│                                      /boot
└─sda2         8:2    0   1.8T  0 part /var
                                       /home
sdb            8:16   1 115.2G  0 disk 
└─sdb1         8:17   1 115.2G  0 part /run/media/gamer/6cc09f63-45e7-4028-9243-3b64b8028dfa
mmcblk0      179:0    0  29.2G  0 disk 
├─mmcblk0p1  179:1    0   511M  0 part 
└─mmcblk0p2  179:2    0  28.7G  0 part /frzr_root
                                       /
mmcblk0boot0 179:8    0     4M  1 disk 
mmcblk0boot1 179:16   0     4M  1 disk 
[gamer@chimeraos 6cc09f63-45e7-4028-9243-3b64b8028dfa]$ df -H

Filesystem      Size  Used Avail Use% Mounted on
dev              16G     0   16G   0% /dev
run              16G  1.1M   16G   1% /run
/dev/mmcblk0p2   31G  8.5G   22G  28% /
/dev/mmcblk0p2   31G  8.5G   22G  28% /frzr_root
overlay          31G  8.5G   22G  28% /etc
tmpfs            16G  176M   16G   2% /dev/shm
tmpfs            16G  730k   16G   1% /tmp
/dev/sda1       536M   92M  445M  18% /boot
/dev/sda2       2.0T  8.3G  2.0T   1% /home
/dev/sda2       2.0T  8.3G  2.0T   1% /var
tmpfs           3.2G  177k  3.2G   1% /run/user/1000
/dev/sdb1       122G  2.2M  115G   1% /run/media/gamer/6cc09f63-45e7-4028-9243-3b64b8028dfa
[gamer@chimeraos 6cc09f63-45e7-4028-9243-3b64b8028dfa]$`

**Comment by pastaq** [comment]
> ```
> 
> NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
> sda            8:0    0   1.8T  0 disk 
> ├─sda1         8:1    0   511M  0 part /frzr_root/boot
> │                                      /boot
> └─sda2         8:2    0   1.8T  0 part /var
>                                        /home
> sdb            8:16   1 115.2G  0 disk 
> └─sdb1         8:17   1 115.2G  0 part /run/media/gamer/6cc09f63-45e7-4028-9243-3b64b8028dfa
> mmcblk0      179:0    0  29.2G  0 disk 
> ├─mmcblk0p1  179:1    0   511M  0 part 
> └─mmcblk0p2  179:2    0  28.7G  0 part /frzr_root
>                                        /
> mmcblk0boot0 179:8    0     4M  1 disk 
> mmcblk0boot1 179:16   0     4M  1 disk 
> [gamer@chimeraos 6cc09f63-45e7-4028-9243-3b64b8028dfa]$ df -H
> 
> Filesystem      Size  Used Avail Use% Mounted on
> dev              16G     0   16G   0% /dev
> run              16G  1.1M   16G   1% /run
> /dev/mmcblk0p2   31G  8.5G   22G  28% /
> /dev/mmcblk0p2   31G  8.5G   22G  28% /frzr_root
> overlay          31G  8.5G   22G  28% /etc
> tmpfs            16G  176M   16G   2% /dev/shm
> tmpfs            16G  730k   16G   1% /tmp
> /dev/sda1       536M   92M  445M  18% /boot
> /dev/sda2       2.0T  8.3G  2.0T   1% /home
> /dev/sda2       2.0T  8.3G  2.0T   1% /var
> tmpfs           3.2G  177k  3.2G   1% /run/user/1000
> /dev/sdb1       122G  2.2M  115G   1% /run/media/gamer/6cc09f63-45e7-4028-9243-3b64b8028dfa
> [gamer@chimeraos 6cc09f63-45e7-4028-9243-3b64b8028dfa]$
> ```

This is really old. /frzr_root, /frzr_boot, and /home should all be on the same drive. The reason it's not mounting is because we check for those directories before mounting them or showing them in the interface so you can't override your root filesystem. I'm not sure how the installer messed this up and split the install onto two drives. I would suggest you wipe both drives, remove sda, reinstall to mmcblk0, then reinstall sda. You'll need to use a live USB of something like EndeavourOS to wipe the drives since both drives will be mounted inside ChimeraOS.

**Comment by daniellasalle** [comment]
By "really old" you mean "really odd"?

Also I have already tried what you suggested (See my post from 3 days ago in this same thread).

By doing what you suggest it will end up the same: I will be able to boot off my 31Gbs mmcblk0p2 but my 2Tbs will not be detected by Chimera OS. So we'll have gotten full circle.

Any other suggestion?

**Comment by pastaq** [comment]
> By "really old" you mean "really odd"?
Yes, my mistake
> Also I have already tried what you suggested (See my post from 3 days ago in this same thread).
> 
> By doing what you suggest it will end up the same: I will be able to boot off my 31Gbs mmcblk0p2 but my 2Tbs will not be detected by Chimera OS. So we'll have gotten full circle.
> 
> Any other suggestion?
What I am suggesting is slightly different. Specifically, physically remove the second drive after formatting it to a blank drive. The reinstall won't be able to touch the second drive. After the initial install,

**Comment by pastaq** [comment]
Sorry, stupid phone glitched out.

**Comment by alkazar** [comment]
Do you have ChimeraOS installed or previously installed on multiple disks?

**Comment by daniellasalle** [comment]
In over 2 years I never managed to get ChimeraOS working out on my Atari VCS which has 2 drives installed.

It always ended up with 1 drive being discovered and useable. Always.

**Comment by daniellasalle** [comment]
> What I am suggesting is slightly different. Specifically, physically remove the second drive after formatting it to a blank drive. The reinstall won't be able to touch the second drive. After the initial os install, reinstall the second drive

I understand that the goal is to reinstall ChimeraOS while the device has 2 drives.

However I do not understand "physically remove the second drive after formatting it to a blank drive".

Why would I need to format it if I am going to end up removing it?

What does "formatting it to a blank drive" means? Does it mean it should have no partition(As I think would be best) or should it have a formatted ext4 partition available?

But then back to my previous question of the drive being ended up removed.

Please educate me on this matter before I move forward with this request.

**Comment by pastaq** [comment]
> > What I am suggesting is slightly different. Specifically, physically remove the second drive after formatting it to a blank drive. The reinstall won't be able to touch the second drive. After the initial os install, reinstall the second drive
> 
> I understood that the goal was to reinstall ChimeraOS while the Atari VCS had 2 drives plugged in. But now I am not so sure anymore mostly because of the "physically remove the second drive after formatting it to a blank drive" part.
> 
> Why would I need to format it if I am going to end up removing it?
> 
> What does "formatting it to a blank drive" means? Does it mean it should have no partition(As I think would be best) or should it have a formatted EXT4 partition available? Or should it be a NTFS? Or it doesn't matter?
> 
> But regardless of this answer I am back to my previous question. If I physically remove the drive before I install ChimeraOS on the first drive why does it matter if my 2nd drive had a partition or what format it was?
> 
> If you could supply me a step by step guide of what you would like to do that would be best to help me understand where we want to go and what we want to do.

The program that mounts the RW overlays uses drive labels rather than drive numbers or UUID due to some historical reasons I won't get into here. Since both the sda and mmcblk drives have a label for the expected mount points, frzr is randomly grabbing one or the other and mounting it. This will prevent the removable media automount scripts from working because they detect one of the restricted mount points we intentionally block to prevent accidentally wiping your main OS drive. We need to clear the labels on both drives before we do anything so that it doesn't grab the wrong one again during boot up after the physical reinstall of the second drive later. You can just delete all the partitions using `parted` from any liveusb for both drives. We will format them later.

Once both drives are blank, physically remove the drive you wish to use as a steam library only so we can ensure frzr can't touch it again during the install. This is probably not completely necessary but more an abundance of caution so we can say for sure it won't be written to during the install. I'd recommend using mmcblk0 as your boot drive and sda as your library drive, but I leave that up to you.

After you have just the one drive installed in the system, perform a full install of the OS. If you cleared the partitions correctly it shouldn't ask you to repair, but if it does just select no. When it asks for normal or advanced install, select advanced then use spacebar to select unstable builds. This will give you the most up to date version of the steam-removable-media utility which has some recent bug fixes. That will ensure any bugs we discover are relevant to the latest version and haven't been squashed yet. If you'd prefer to stay on stable that will work too, but you'll need to do some extra steps which I will detail below.

Boot into the system one time to ensure install went correctly and run through the OOBE until you've logged in to steam for the first time. Now would be a good time to set up SSH. If you opted to remain on stable then you need to unlock the system with`frzr-unlock` and update the package with `pikaur -Sy steam-removable-media-git` (both commands as `sudo`). Shut down the system and reinstall the second drive. After you boot up again we'll need to check the drive utility in the chimera web app to see if the drive is detected. If it is, select "format drive" in the UI, wait a while, and check in steam for the drive once success is reported. If any of this part fails, post the results of `df -h` and `lsblk` again so we can verify the drive is detected and look for anomalies.

You can also try manually running the script with `/usr/bin/shadowblip/format-media --full --device /dev/sda` and post any errors that pop up.

If the format works successfully but you still have no library in steam, `journalctl -u media-mount@sda1` should provide a log of the mounting.

**Comment by pastaq** [comment]
Have you had a chance to attempt this yet?

**Comment by pastaq** [comment]
Does the device show up in storage after having mounted it from the CLI or is it still blank?

**Comment by daniellasalle** [comment]
The new device does show up under Storage and was automatically mounted under ```/run/media/gamer/deviceUUIDhere``` after it was formatted, yes.

**Comment by pastaq** [comment]
Cool. Curious that it didn't show the first time.

**Comment by daniellasalle** [comment]
Is there anything else I could do before we get to joyfully mark this one as complete?

**Comment by pastaq** [comment]
Seems good to me, but I'm not a user of that device. You can close it if it's working for you.

**Comment by daniellasalle** [comment]
@photoncody the ball' is in your camp buddy

**Comment by yhilgjff** [comment]
NB: After applying this and rebooting, there was an update and then when installing a game it pointed back on the 1st drive. I then selected the 2nd drive has the starred one and it worked to this very day.
